Automattic/vip-decoupled-bundle
# Add the block data api plugin

## Description

The purpose of this PR is to add the block data api plugin into the bundle. Since the block data api plugin has a rest and graphQL endpoint, this will now enable both of the APIs with this bundle.

The eventual goal would be to replace the bespoke the `WPGraphQL Content blocks` plugin, in favour of this plugin. The first step is add the plugin in, and then slowly add in any missing features so we can eventually deprecate it.

To review this PR, the changes that have been made:

- Added the plugin using `git subtree`:

`git remote add -f block-data-api-remote git@github.com:Automattic/vip-block-data-api.git`
`git subtree add --prefix lib/vip-block-data-api block-data-api-remote trunk --squash`

The following is how you update it:

`git fetch block-data-api-remote trunk`
`git subtree pull --prefix lib/vip-block-data-api block-data-api-remote trunk --squash`

- Added the block data api as a setting that can be turned on and off within the `settings.php`
- Required the plugin if its enabled within the `vip-decoupled.php`
- Updated the README with this new plugin with potential deprecation plans.

## Steps to Test

- Load up the dev-env or wp-env and enable this plugin
- Ensure the graphQL endpoint can be hit, using the query:

```gql
query AllPosts {
  posts {
    nodes {
      id
      title
      contentBlocks {
        blocks {
          attributes {
            name
            value
          }
          name
          innerHTML
        }
        isGutenberg
        version
      }
    }
  }
}
```

- Ensure the `wp-json/vip-block-data-api/v1/posts/1/blocks` rest endpoint can be hit
